### ISSUE TITLE
[Accessibility] Add fieldset and legend to payment step

### ISSUE THREAD

**Issue by gazjoy** [Adyen/adyen-web#1711]
**Describe the bug**
When reviewing our checkout for [WCAG 2.1 - 1.3.1 Info and Relationships (A)](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html), I found the payment step not to have any `<fieldset>` or `<legend>` tags. These elements allows users to understand the relationship of the controls and interact with the form more quickly and effectively.

**To Reproduce**
1. Go to payment step
2. View payment step HTML
3. See no fieldset or legend

**Expected behavior**
We need to add `<fieldset>` with direct descendant of `<legend>` to the payment step. These elements work together to tell screen readers that a group of form fields relate to each other, and to provide a label for the group.

**Code example**

```
<div>
  <div><h2>Payment</h2></div>
  <fieldset>
    <legend>Payment type</legend>
    <div>
      <input id="pay-cc" name="payment-type" type="radio" value="cc">
      <label for="pay-cc">Credit Card</label>
    </div>
    <div>
      <input id="pay-pp" name="payment-type" type="radio" value="pp">
      <label for="pay-pp">PayPal</label>
    </div>
  </fieldset>
  <fieldset>
    <legend>Card details</legend>
    <div>
      <input id="name-cc" name="payment-type" type="text" value="name">
      <label for="name-cc">Name on card</label>
    </div>
<!-- Additional fields -->
  </fieldset>
</div>
```

**Screenshots**
<img width="692" alt="image" src="https://user-images.githubusercontent.com/628064/186461991-cf530424-e7ab-4dcb-8432-908cd990f4c9.png">

**Desktop (please complete the following information):**
 - OS: Mac OS Monterey
 - Browser Safari 15

**Additional context**
Further reading on why this is important. https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/

**Comment by sponglord** [comment]
Hi @gazjoy - thanks for this observation. 
As mentioned previously we will wait for the findings of our a11y assessment before implementing such structural changes

**Comment by sponglord** [comment]
Our a11y assessment did flag that `fieldsets` and `legends` were missing from our markup, _but not_ in relation to the payment method list.

It did however flag that they were needed elsewhere - and so we added them as part of this PR (#2083)
